service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendAcknowledgementTaskDispatcher.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SendAcknowledgementTaskDispatcher {
+    private final TaskDispatcher taskDispatcher;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public void send(EhrExtractStatus ehrExtractStatus, String typeCode) {
+        var sendAcknowledgementTaskDefinition = SendAcknowledgementTaskDefinition.builder()
+            .taskId(randomIdGeneratorService.createNewId())
+            .conversationId(ehrExtractStatus.getConversationId())
+            .requestId(ehrExtractStatus.getEhrRequest().getRequestId())
+            .toAsid(ehrExtractStatus.getEhrRequest().getToAsid())
+            .fromAsid(ehrExtractStatus.getEhrRequest().getFromAsid())
+            .fromOdsCode(ehrExtractStatus.getEhrRequest().getFromOdsCode())
+            .toOdsCode(ehrExtractStatus.getEhrRequest().getToOdsCode())
+            .nhsNumber(ehrExtractStatus.getEhrRequest().getNhsNumber())
+            .typeCode(typeCode)
+            .build();
+
+        taskDispatcher.createTask(sendAcknowledgementTaskDefinition);
+        LOGGER.info(String.format("SendAcknowledgementTaskDefiniiton added to task queue, conversationId: %s",
```

Comment:
You don't need to log conversationId: %s here, it is always included in our logs via the logback format / MDC

Reply:
also typo SendAcknowledgementTaskDefiniiton